keeper/msg_server.go

```diff
@@ -194,6 +197,12 @@ func (k *Keeper) UpdateIndex(ctx context.Context, index int64) error {
 		return err
 	}
 
+	// Claim and transfer the yield of hyperlane external chains.
+	err = k.claimExternalYieldHyperlane(ctx)
```

Comment:
nit: I'm always in favor of better function names instead of rely on comments to explain a function. Here is easy:

```suggestion
	err = k.claimAndTransferExternalYieldHyperlane(ctx)
```

Reply:
I agree with this, but to be more aligned with the rest of the module (like vaults), let's leave this as is for now!